### PR TITLE
[finishes #175743307] configuring unit tests

### PR DESCRIPTION
**Pull request by leny62**
##Description
Configuration of unit tests

##Description of Task to be completed?
This branch contains files that configures unit testing using Mocha as the test runner with also chai and chai-http for the unit tests.

##How should this be manually tested?
*Clone this repo
*Open the terminal or git bash
*write npm i 
and lastly 
*write npm test/ npm run test

##What are the relevant pivotal tracker stories?
Setup unit testing [test](https://www.pivotaltracker.com/story/show/175743307) #175743307

### CODE REVIEW

File: src/tests/simple.test.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    let result= add(5);
+    assert.typeOf(result, 'number');
+  })
+})
+
+describe('First', () =>{
+  it('should return Hello world', ()=>{
+    assert.equal(first(), 'Hello world');
+  } )
+} )
```
Comment (Anguandia) [inline src/tests/simple.test.js:21]: You have not tested any HTTP request though

File: src/app.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+.then(() => console.log('connected to mongodb successfully....'))
+.catch(err =>console.log('Failed to connect to mongodb!',err));
+
+app.use(json());
+app.use(urlencoded({extended: true}))
+
+
+app.get('/',(req,res)=>{
+    res.send("Welcome to my app").status(200)
+});
```
Comment (Anguandia) [inline src/app.js:23]: You need to test this route

File: src/tests/simple.test.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { assert } from 'chai';
+
+import add from '../controllers/add';
+import first from '../controllers/first';
```
Comment (Anguandia) [inline src/tests/simple.test.js:4]: What happened to the test client?
Please create and export a general setting for tests that will be imported in all subsequent test files

File: test.js [hunk 4]
```diff
@@ -0,0 +1 @@
+var foo = bar;
```
Comment (Anguandia) [inline test.js]: ??? is this relevant?

**Review by Anguandia — CHANGES_REQUESTED** [review]
First.js and add.js are not really necessary, please test the index route instead

File: test.js (hunk 4 above)
Comment (Anguandia) [reply test.js]: Also squash the commits

File: src/app.js (hunk 2 above)
Comment (leny62) [reply src/app.js:23]: Noted!

File: test.js (hunk 4 above)
Comment (leny62) [reply test.js]: This was a simple file to test Eslint

**Pull request closed without merging**